Lesson 05/lesson05 script2.js

```diff
@@ -0,0 +1,259 @@
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const minValue = arr.reduce((a, b) => Math.min(a, b));
+
+
+// console.log(`The minimum element of an array is: ${minValue} 
+// Index of the min element is: `+ arr.indexOf(minValue));
+
+
+
+
+
+// 3. Знайти максимальний елемент масиву та його порядковий номер.
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const maxValue = arr.reduce((a, b) => Math.max(a, b), -Infinity);
+
+
+// console.log(`The maximum element of an array is: ${maxValue} 
+// Index of the max element is: `+ arr.indexOf(maxValue));
+
+
+
+
+
+
+
+// 4. Визначити кількість негативних елементів.
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithNegValues = arr.filter((elem) => {
+//     if(elem < 0) {
+//     return elem
+// } 
+// } )
+
+// console.log(ArrayWithNegValues.length)
+
+
+
+
+
+// 5.Знайти кількість непарних позитивних елементів.
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithOddPosValues = arr.filter((elem) => {
+//     if(elem > 0 && elem % 2 !== 0){
+//     return true
+
+// } 
+// } )
+
+// console.log(ArrayWithOddPosValues.length)
+
+
+
+
+
+
+
+// 6. Знайти кількість парних позитивних елементів.
+//done
+
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithEvenPosValues = arr.filter((elem) => {
+//     if(elem > 0 && elem % 2 === 0){
+//     return true
+// } 
+// } )
+
+// console.log(ArrayWithEvenPosValues.length)
+
+
+
+
+
+
+
+
+
+// 7. Знайти суму парних позитивних елементів.
+// done 
+
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithEvenPosValues = arr.filter((elem) => {
+//     if(elem >= 0 && elem % 2 === 0){
+//     return true
+// } 
+// } )
+// console.log(ArrayWithEvenPosValues)
+
+
+// const result = ArrayWithEvenPosValues.reduce((total, elem) => {
+//     return total + elem            
+// }, 0) 
+
+//  console.log(result);
+
+
+
+
+
+
+//  8. Знайти суму непарних позитивних елементів.
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithOddPosValues = arr.filter((elem) => {
+//     if(elem >= 0 && elem % 2 !== 0){
+//     return true
+// } 
+// } )
+// console.log(ArrayWithOddPosValues)
+
+
+// const amount = ArrayWithOddPosValues.reduce((total, elem) => {
+//     return total + elem            
+// }, 0) 
+
+//  console.log(amount);
+
+
+
+
+
+
+
+// 9. Знайти добуток позитивних елементів.
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithOddPosValues = arr.filter((elem) => {
+//     if(elem >= 0){
+//     return true
+// } 
```

Comment:
можна написати коротше
```suggestion
//     return elem >= 0;
```